### DISCUSSION TITLE
comparison to Gradio

### DISCUSSION THREAD

**Discussion by Borda** [zauberzeug/nicegui#3020] [Q&A]
Hello, in the README, I found some nice explanations/comparisons to Streamlit, so I would like to hear if you have a similar comparison also to the other very popular GUI library - https://www.gradio.app

**Comment by stdusr** [comment]
Here  is a comparison of multiple UI frameworks: (Anvil, Panel, Dash, NiceGUI, IDOM, Epyk, Flet, Pynecode, Lona, Nagare, Reahl and many more)
https://www.may69.com/purepython/#No_thanks_Streamlit

**Comment by rodja** [comment]
We have not looked into Gradio but would be happy if someone could shed some light onto the differences.

**Reply by noontee0** [reply]
Gradio is very abstract and seems to only be able to do single page web applications. I haven't even found any documentation about routing. It is mainly used to quickly build AI form based web applications
